src/main/java/crud/dao/impl/DriverDaoJdBcImpl.java

```diff
@@ -0,0 +1,115 @@
+            throw new DataBaseException("Couldn't create insert "
+                    + driver + " into driversDB. ", throwables);
+        }
+    }
+
+    @Override
+    public Optional<Driver> get(Long id) {
+        String query = "SELECT driver_id, driver_name, driver_license_number FROM drivers "
+                + "WHERE driver_id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet set = statement.executeQuery();
+            Driver driver = null;
+            if (set.next()) {
+                driver = setDriver(set);
+            }
+            return Optional.ofNullable(driver);
+        } catch (SQLException throwables) {
+            throw new DataBaseException("Couldn't get driver by id " + id, throwables);
+        }
+    }
+
+    @Override
+    public List<Driver> getAll() {
+        String query = "SELECT * FROM drivers WHERE deleted = FALSE";
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet set = statement.executeQuery();
+            while (set.next()) {
+                drivers.add(setDriver(set));
+            }
+            return drivers;
+        } catch (SQLException throwables) {
+            throw new DataBaseException("Couldn't get a list of drivers from driversDB.",
+                    throwables);
+        }
+    }
+
+    @Override
+    public Driver update(Driver driver) {
+        String query = "UPDATE drivers "
+                + "SET driver_name = ?, driver_license_number = ? "
+                + "WHERE driver_id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = setUpdate(connection.prepareStatement(query), driver)) {
+            statement.setLong(3, driver.getId());
+            statement.executeUpdate();
+            return driver;
+        } catch (SQLException throwables) {
+            throw new DataBaseException("Couldn't update " + driver + " in driversDB.", throwables);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE drivers SET deleted = TRUE WHERE driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException throwables) {
+            throw new DataBaseException("Couldn't delete driver with id " + id, throwables);
+        }
+    }
+
+    static Driver setDriver(ResultSet resultSet) throws SQLException {
```

Comment:
let's better repeat this code in each class then make this method static and without access modifier